client/pong.c

```diff
@@ -312,11 +350,16 @@ void pong_run(Pong* pong) {
   pong->running = true;
 
   while (pong->running) {
-    // process events
-    game_step_begin(&pong->game);
-    pong_process_events(pong);
-    game_step_end(&pong->game, TICK_MS);
-
+    if (pong->connection_state.state == LOCAL) {
```

Comment:
That's not correct. We should not have different implementation of entire game loop depending on connection state. The events should be handled always the same way. With your changes user will not be able to stop moving after he pressing <- or ->.